mod/gc_communities/views/default/page/layouts/index.php

```diff
@@ -34,24 +34,46 @@
 
 <div class="row">
     <div class="col-md-8">
-        <?php
-
+        <?php      
         $options = array(
-                'type' => 'object',
-                'subtypes' => $subtypes,
-                'limit' => $newsfeed_limit,
-                'metadata_name' => 'audience',
-                'metadata_values' => $community_audience, //We can add allps metadata here
-                'full_view' => false,
-                'list_type_toggle' => false,
-                'pagination' => true
-            );
-
+            'type' => 'object',
+            'subtypes' => $subtypes,
+            'limit' => $newsfeed_limit,
+            'full_view' => false,
+            'list_type_toggle' => false,
+            'pagination' => true
+            
+        );
+        $tags_values ='';
+        $tags_display ='';
+        if( strpos($community_tags, ',') !== false ){ // if multiple tags
+                $community_tags = array_map('trim', explode(',', $community_tags));
+            foreach($community_tags as $tag_val){
+                $tags_value = elgg_get_metastring_id($tag_val);
+                $tags_values .= ' OR md.value_id = '.$tags_value;
+                $tags_display .= ' <span class="elgg-tag" style="color:#055959; font-size:12px; padding:3px 8px;">'.$tag_val.'</span>';
+            }
+        }else{
+           $tags_values = ' OR md.value_id = ' .elgg_get_metastring_id($community_tags);
+            $tags_display = '<span class="elgg-tag" style="color:#055959; font-size:12px; padding:3px 8px;">' .$community_tags .'</span>';
+        }
+        //get the metastring id
+        $audience_name = elgg_get_metastring_id('audience');
+        $audience_value = elgg_get_metastring_id($community_audience);
+        $tags_name = elgg_get_metastring_id('tags');
+        //Query grabs content with audience or tags
+        $dbprefix = elgg_get_config('dbprefix');
+        $options['joins'][] ="JOIN {$dbprefix}metadata md ON (e.guid = md.entity_guid)";
+        $options['joins'][] ="JOIN {$dbprefix}metastrings ms ON (ms.id = md.value_id)";
+        $options['joins'][] ="JOIN {$dbprefix}metastrings ms1 ON (ms1.id = md.name_id)";

```

Comment:
The joined metastrings aren't queried, so you don't need to join them.

Reply:
good call 👌